src/Resources/skeleton/model/Repository.tpl.php

```diff
@@ -0,0 +1,64 @@
+<?= "<?php\n" ?>
+
+namespace <?= $namespace; ?>;
+
+<?= $use_statements ?>
+
+class <?= $class_name ?> extends ServiceEntityRepository
+{
+    /**
+    * @extends ServiceEntityRepository<<?= $entity_class_name ?>>
+    *
+    * @method <?= $entity_class_name ?>|null find($id, $lockMode = null, $lockVersion = null)
+    * @method <?= $entity_class_name ?>|null findOneBy(array $criteria, array $orderBy = null)
+    * @method <?= $entity_class_name ?>[]    findAll()
+    * @method <?= $entity_class_name ?>[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+    */
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, <?= $entity_class_name ?>::class);
+    }
+
+    public function save(<?= $entity_class_name ?> $entity, bool $flush = false): void
+    {
+        $this->getEntityManager()->persist($entity);
+
+        if ($flush) {
+            $this->getEntityManager()->flush();
+        }
+    }
+
+    public function remove(<?= $entity_class_name ?> $entity, bool $flush = false): void
+    {
+        $this->getEntityManager()->remove($entity);
+
+        if ($flush) {
+            $this->getEntityManager()->flush();
+        }
+    }
+
+//    /**
+//     * @return <?= $entity_class_name ?>[] Returns an array of <?= $entity_class_name ?> objects
+//     */
+//    public function findByExampleField($value): array
+//    {
+//        return $this->createQueryBuilder('t')
+//            ->andWhere('t.exampleField = :val')
+//            ->setParameter('val', $value)
+//            ->orderBy('t.id', 'ASC')
+//            ->setMaxResults(10)
+//            ->getQuery()
+//            ->getResult()
+//        ;
+//    }
+
+//    public function findOneBySomeField($value): ?<?= $entity_class_name ?>
+//    {
+//        return $this->createQueryBuilder('t')
+//            ->andWhere('t.exampleField = :val')
+//            ->setParameter('val', $value)
+//            ->getQuery()
+//            ->getOneOrNullResult()
+//        ;
+//    }
```

Comment:
**nitpick**: I like the idea of included sample code!

There's an alternative pattern used by our repository, which is a bit confusing at first look, but is using a more FP-style approach:

```php
public function findByExampleField($value): static
{
    return $this->filter(function(QueryBuilder $queryBuilder) use ($value) {
        $queryBuilder
            ->andWhere(self::TABLE_ALIAS . '.exampleField = :val')
            ->setParamter('val', $value)
            ->orderBy(self::TABLE_ALIAS . '.id', 'ASC')
        ;
    });
}
```

- The `filter` method creates a clone and then modifies the clone's internal `QueryBuilder` instead of the original one.
- The repository class itself implements `\Iterator`, so you can `foreach (...)` through its results
- By returning itself, you can do method chaining

With this pattern it's possible to write code like this:

```php
$fooArticles = $articlesRepo->findByFoo();

foreach ($fooArticles as $fooArticle) { ... }

$paginatedBarArticles = $articlesRepo->findbyBar()->paginate(10, 1);

foreach ($paginatedBarArticles as $barArticle) { ... }
```

Would be cool if we could adjust the sample code for that pattern 😊